app/scripts/controllers/network/provider-api-tests/helpers.js

```diff
@@ -286,14 +295,23 @@ export async function withMockedInfuraCommunications(...args) {
  * on that object.
  * @returns {Promise<any>} The return value of the given function.
  */
-export async function withInfuraClient(...args) {
+export async function withClient(...args) {
   const [options, fn] = args.length === 2 ? args : [{}, args[0]];
-  const { network = 'mainnet' } = options;
+  const { network = 'mainnet', type = 'infura' } = options;
 
-  const { networkMiddleware, blockTracker } = createInfuraClient({
-    network,
-    projectId: INFURA_PROJECT_ID,
-  });
+  let clientUnderTest;
+  if (type === 'infura') {
+    clientUnderTest = createInfuraClient({
+      network,
+      projectId: INFURA_PROJECT_ID,
+    });
+  } else {
+    clientUnderTest = createJsonRpcClient({
+      rpcUrl: 'http://localhost:8545',
+      chain_id: '0x1',
```

Comment:
another note, kind of surprising that this doesn't make some tests fail (some tests are using different networks and checking their chain id...)

Reply:
should this be `chainId`

Reply:
Hmm. I'll take a closer look at this.

Reply:
Ive fixed this issue actually. There was some code that wasnt passing in type. This would have been caught if it was required, so + points to that suggestion :)

Reply:
Oh ok cool!